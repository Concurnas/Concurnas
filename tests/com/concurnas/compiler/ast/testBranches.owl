//if elif else
if(x > 4){print(x);}
else{ print(4);}

if(x > 4){print(x);}
elif (x < 5){	print(33);  }
elif (x < 99){	print(33);  }
else{ print(4);}

//while
while(false)
{
	print(d);
	continue;
	break;
}

//for
for(a in [1,2,3])
{ print(a); }

for(a int in [1,2,3])
{ print(a); }

//try catch
try{a = 5;}
catch(Exception e){print(e);}catch(Exception2 e){print(e);}
finally{print("done");}

//scope and async block
{x int = 5;}
{x int = 5;}!

//throw return
def a() void
{
	if(false) { return;}
	elif(nice) { return 55;}
	else{return 55;}
	throw new Exception("aa");
}


if(a==5 or b ==6)
{
	g = 8 
}



ar = [1,2,3,4]

def doings() {

	x = parfor(a in ar){
		a + 10
	}
	x2 = parforsync(a in ar){
		a + 10
	}
	
	x3 = parfor(n=0; n< ar.length; n++){
		ar[n] + 10
	}
	x4 = parforsync(n=0; n< ar.length; n++){
		ar[n] + 10
	}

	"hi: " + [x, x2, x3, x4]
}