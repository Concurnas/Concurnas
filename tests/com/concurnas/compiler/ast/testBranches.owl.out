{
if ( x > 4 ) {
print ( x ) ;
}
else {
print ( 4 ) ;
}
if ( x > 4 ) {
print ( x ) ;
}
elif ( x < 5 ) {
print ( 33 ) ;
}
elif ( x < 99 ) {
print ( 33 ) ;
}
else {
print ( 4 ) ;
}
while ( false ) {
print ( d ) ;
continue ;
break ;
}
for ( a in [ 1 , 2 , 3 ] ) {
print ( a ) ;
}
for ( a int in [ 1 , 2 , 3 ] ) {
print ( a ) ;
}
try {
a = 5 ;
}
catch ( Exception e ) {
print ( e ) ;
}
catch ( Exception2 e ) {
print ( e ) ;
}
finally {
print ( " done " ) ;
}
{
x int = 5 ;
}
{
x int = 5 ;
}
! def a ( ) void {
if ( false ) {
return ;
}
elif ( nice ) {
return 55 ;
}
else {
return 55 ;
}
throw new Exception ( " aa " ) ;
}
if ( a == 5 or b == 6 ) {
g = 8 ;
}
ar = [ 1 , 2 , 3 , 4 ] ;
def doings ( ) void {
x = for ( a in ar ) {
{
a + 10 ;
}
! }
;
x2 = try {
com \. concurnas \. bootstrap \. runtime \. cps \. Fiber \. getCurrentFiber ( ) \. enterSync ( ) ;
for ( a in ar ) {
{
a + 10 ;
}
! }
}
finally {
com \. concurnas \. bootstrap \. runtime \. cps \. Fiber \. getCurrentFiber ( ) \. exitsync ( ) ;
}
;
x3 = for ( n = 0 ; n < ar . length ; n ++ ) {
{
ar [ n ] + 10 ;
}
! }
;
x4 = try {
com \. concurnas \. bootstrap \. runtime \. cps \. Fiber \. getCurrentFiber ( ) \. enterSync ( ) ;
for ( n = 0 ; n < ar . length ; n ++ ) {
{
ar [ n ] + 10 ;
}
! }
}
finally {
com \. concurnas \. bootstrap \. runtime \. cps \. Fiber \. getCurrentFiber ( ) \. exitsync ( ) ;
}
;
" hi:  " + [ x , x2 , x3 , x4 ] ;
}
}